RealmSwift/Object.swift

```diff
@@ -219,33 +219,23 @@ public func == <T: Object>(lhs: T, rhs: T) -> Bool {
 
 /// Object interface which allows untyped getters and setters for Objects.
 public final class DynamicObject : Object {
-    private var listProperties = [String: List<DynamicObject>]()
+    private var listProperties = [String: Optional<List<DynamicObject>>]()
 
-    /// Returns or sets the value of the property with the given name.
-    public override subscript(key: String) -> AnyObject? {
-        get {
-            if let prop = RLMObjectBaseObjectSchema(self)[key] {
-                if prop.type == .Array {
-                    // get it or set it
-                    if let list = listProperties[key] {
-                        return list
-                    }
-                    let list = List<DynamicObject>()
-                    listProperties[key] = list
-                    return list
-
-                }
-            }
-            return RLMObjectBaseObjectForKeyedSubscript(self, key)
+    private override func listProperty(key: String) -> RLMListBase? {
+        // get it or set it
+        if let list = listProperties[key] {
```

Comment:
We want to return if `listProperties` contains the key, not just if the result of the subscript isn't nil. As it is now, this optional binding will fail even when we've already determined that the given key isn't a List.

```swift
if contains(listProperties.keys, key) {
  return listProperties[key]
}
```